docs/feature-flagging/randomization-sdk/client-sdks/javascript.md

```diff
@@ -0,0 +1,156 @@
+# JavaScript
+
+:::note
+This documentation is for our client-side SDK intended for use by browser applications. For use in server applications, refer to our [Node JS SDK](../server-sdks/node.md).
+:::
+
+### 1. Install the SDK
+You can install the SDK with Yarn or NPM:
+
+<Tabs>
+<TabItem value="yarn" label="Yarn">
+
+```bash
+yarn add @eppo/js-client-sdk
+```
+
+</TabItem>
+
+
+<TabItem value="npm" label="NPM">
+
+```bash
+npm install @eppo/js-client-sdk
+```
+
+</TabItem>
+</Tabs>
+
+### 2. Define an Assignment Logger
+
+The SDK requires an assignment logger to be passed on initialization. The SDK invokes the logger to capture assignment data whenever a variation is assigned. The below code example shows how to integrate the SDK with [Segment](https://segment.com/docs/) for logging events. You could also use your own logging system; the only requirement is that the SDK receives a `logAssignment` function.
+
+Define an implementation of the Eppo `AssignmentLogger` interface. This interface has one function: `logAssignment`.
+
+```javascript
+import { IAssignmentLogger } from '@eppo/js-client-sdk';
+import { AnalyticsBrowser } from '@segment/analytics-next'
+
+// Connect to Segment (or your own event-tracking system)
+const analytics = AnalyticsBrowser.load({ writeKey: '<SEGMENT_WRITE_KEY>' })
+
+const assignmentLogger: IAssignmentLogger = {
+  logAssignment(assignment) {
+    analytics.track({
+      userId: assignment.subject,
+      event: 'Eppo Randomized Assignment',
+      type: 'track',
+      properties: { ...assignment }
+    });
+  },
+};
+```
+
+The SDK will invoke the `logAssignment` function with an `assignment` object that contains the following fields:
+
+| Field | Description | Example |
+| --------- | ------- | ---------- |
+| `experiment` (string) | An Eppo experiment key | "recommendation_algo" |
+| `subject` (string) | An identifier of the subject or user assigned to the experiment variation | UUID |
+| `variation` (string) | The experiment variation the subject was assigned to | "control" |
+| `timestamp` (string) | The time when the subject was assigned to the variation | 2021-06-22T17:35:12.000Z |
+| `subjectAttributes` (map) | A free-form map of metadata about the subject. These attributes are only logged if passed to the SDK assignment function | `{ "country": "US" }` |
+
+### 3. Initialize the SDK
+
+Initialize the SDK with the event logger from the previous section and your API key.:
+
+```javascript
+import { init } from '@eppo/js-client-sdk';
+
+await init({
+  apiKey: '<API_KEY>',
+  assignmentLogger,
+});
+```
+
+The `init` method downloads your Eppo experiment configurations once per browser session.
+
+:::note
+API Keys used with Client SDKs should have only ‘Randomization READ’ permissions
+:::
+
+### 4. Assign Experiment Variations 
+The SDK returns an assignment based on the experiments you configure in Eppo. It may take up to 10 minutes for changes to Eppo experiments to be reflected by the SDK assignments.
```

Comment:
I thought it was 5 minutes. Is my understanding dated or is 10 minutes adding buffer?

Reply:
The cache control header is 10 minutes on [this line](https://github.com/Eppo-exp/eppo/blob/main/packages/api/src/randomized-assignment/randomized-assignment.controller.ts#L18)